extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/staticmethods/InterceptedStaticMethodsProcessor.java

```diff
@@ -0,0 +1,495 @@
+
+import io.quarkus.arc.InjectableInterceptor;
+import io.quarkus.arc.deployment.BeanArchiveIndexBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.arc.deployment.BeanRegistrationPhaseBuildItem;
+import io.quarkus.arc.deployment.InterceptorResolverBuildItem;
+import io.quarkus.arc.impl.CreationalContextImpl;
+import io.quarkus.arc.impl.InterceptedMethodMetadata;
+import io.quarkus.arc.impl.InterceptedStaticMethods;
+import io.quarkus.arc.impl.InterceptedStaticMethods.InterceptedStaticMethod;
+import io.quarkus.arc.processor.AnnotationLiteralProcessor;
+import io.quarkus.arc.processor.BeanProcessor;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.arc.processor.InterceptorInfo;
+import io.quarkus.arc.processor.MethodDescriptors;
+import io.quarkus.arc.runtime.InterceptedStaticMethodsRecorder;
+import io.quarkus.deployment.GeneratedClassGizmoAdaptor;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Produce;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveMethodBuildItem;
+import io.quarkus.deployment.util.AsmUtil;
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.DescriptorUtils;
+import io.quarkus.gizmo.FunctionCreator;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+
+public class InterceptedStaticMethodsProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(InterceptedStaticMethodsProcessor.class);
+
+    static final MethodDescriptor INTERCEPTED_STATIC_METHODS_REGISTER = MethodDescriptor
+            .ofMethod(InterceptedStaticMethods.class, "register", void.class, String.class, InterceptedStaticMethod.class);
+    static final MethodDescriptor INTERCEPTED_STATIC_METHODS_AROUND_INVOKE = MethodDescriptor
+            .ofMethod(InterceptedStaticMethods.class, "aroundInvoke", Object.class, String.class, Object[].class);
+
+    private static final String ORGINAL_METHOD_COPY_SUFFIX = "_orig";
+    private static final String INITIALIZER_CLASS_SUFFIX = "_InterceptorInitializer";
+
+    @BuildStep
+    void collectInterceptedStaticMethodsCandidates(BeanArchiveIndexBuildItem beanArchiveIndex,
+            BuildProducer<InterceptedStaticMethodBuildItem> interceptedStaticMethods,
+            InterceptorResolverBuildItem interceptorResolver) {
+
+        // In this step we collect all intercepted static methods, ie. static methods annotated with interceptor bindings  
+        IndexView index = beanArchiveIndex.getIndex();
+
+        for (ClassInfo clazz : index.getKnownClasses()) {
```

Comment:
Wouldn't it be more efficient to query the index for interceptor binding (annotation) uses instead of checking every known class and every known method?

Reply:
+1

I'll update the PR tomorrow!

Reply:
@manovotn done...